sqlmodel/engine/result.py

```diff
@@ -1,29 +1,29 @@
-from typing import Generic, Iterator, List, Optional, TypeVar
+from typing import Generic, Iterator, List, Optional, Sequence, TypeVar
 
 from sqlalchemy.engine.result import Result as _Result
 from sqlalchemy.engine.result import ScalarResult as _ScalarResult
 
 _T = TypeVar("_T")
 
 
-class ScalarResult(_ScalarResult, Generic[_T]):
-    def all(self) -> List[_T]:
+class ScalarResult(_ScalarResult[_T], Generic[_T]):
+    def all(self) -> Sequence[_T]:
         return super().all()
 
-    def partitions(self, size: Optional[int] = None) -> Iterator[List[_T]]:
+    def partitions(self, size: Optional[int] = None) -> Iterator[Sequence[_T]]:
         return super().partitions(size)
 
-    def fetchall(self) -> List[_T]:
+    def fetchall(self) -> Sequence[_T]:
         return super().fetchall()
 
-    def fetchmany(self, size: Optional[int] = None) -> List[_T]:
+    def fetchmany(self, size: Optional[int] = None) -> Sequence[_T]:
```

Comment:
When I started this PR it was the latest version, I haven't tested it with more recent versions but will do soon.

The change from `List` to `Sequence` is because it inherits from sqlalchemy's `ScalarResult` and these are the types defined in the new version so they must be the same in sqlmodel